### ISSUE TITLE
[babel 6] babel-plugin-syntax-async-generators changes output code and generates wrong syntax for async generators defined inside class

### ISSUE THREAD

**Issue by vlasenko** [babel/babel#8382]
## Bug Report
1. Install these packages:
`npm i -S babel-core babel-plugin-syntax-async-generators`
2. Run this code in Node shell:
```js
console.log(require('babel-core').transform(
`
class SearchCriteria {
  async * results (authManager) {  }
}
`,
{ plugins: ["babel-plugin-syntax-async-generators"] }
).code);
```

The output will be:
```
class SearchCriteria {
  *async results(authManager) {}
}
```

Note where the asterisk appeared - its position changed. It is now before `async`, not after `async` as it was in original code. And the code is now broken because of the wrong position of asterisk.

**Environment**
- Babel version(s): v6.26.3
- Node/npm version: Node 8.11.2/npm 5.6.0
- OS: Ubuntu 16.04 LTS
- Monorepo no
- How you are using Babel: `core`

**Comment by xtuc** [comment]
Fixed by https://github.com/babel/babel/pull/6998 and available in Babel 7.

**Comment by vlasenko** [comment]
@xtuc Thank you for following up. Any plans to have this fix backported to Babel 6?

**Comment by xtuc** [comment]
Sorry we will unlikely backport this fix in Babel 6.

**Comment by vlasenko** [comment]
@xtuc Thanks for the answer. According to this page
https://babeljs.io/versions
the current stable version of Babel is version 6.

What if someone backports the fix from Babel 7 to Babel 6, will the team consider merging the fix?

**Comment by vlasenko** [comment]
@xtuc I have created PR which backports the fix to Babel 6: #8383
